idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/purity/PurityPlugin.kt

```diff
@@ -27,7 +29,7 @@ val IdeMetaPlugin.purity: Plugin
         isApplicable = { f: KtNamedFunction ->
           f.nameIdentifier != null && !f.hasModifier(KtTokens.SUSPEND_KEYWORD) &&
             f.resolveToDescriptorIfAny()?.run {
-              !isSuspend && (ReturnsCheck.ReturnsUnit.check(this) || ExtendedReturnsCheck.ReturnsNothing.check(this)
+              !isSuspend && !isSynthesized && !isSuspendLambdaOrLocalFunction() && (ReturnsCheck.ReturnsUnit.check(this) || ExtendedReturnsCheck.ReturnsNothing.check(this)
```

Comment:
@raulraja Should we also make a distinction for `get`, `set`, `init` functions?  As we did back then 
https://github.com/arrow-kt/arrow-meta-prototype/blob/32554a5ade4b49737cfe7ed1e5112415ea84be02/compiler-plugin/src/main/java/arrow/plugin/MetaClassBuilderInterceptorExtension.kt#L54

Reply:
I think so

Reply:
I am not sure either, let's revisit this another time.